controller/src/main/java/org/jboss/as/controller/AttributeMarshaller.java

```diff
@@ -105,6 +105,33 @@ protected String asString(ModelNode value) {
         }
     }
 
+    private static class ObjectMarshaller extends DefaultAttributeMarshaller {
+        private final boolean marshallSimpleTypeAsAttribute;
+
+        private ObjectMarshaller(boolean marshallSimpleTypeAsAttribute) {
+            this.marshallSimpleTypeAsAttribute = marshallSimpleTypeAsAttribute;
+        }
+
+        @Override
+        public void marshallAsElement(AttributeDefinition attribute, ModelNode resourceModel, boolean marshallDefault, XMLStreamWriter writer) throws XMLStreamException {
+            if (resourceModel.hasDefined(attribute.getName()) && attribute instanceof ObjectTypeAttributeDefinition) {
+                AttributeDefinition[] valueTypes = ((ObjectTypeAttributeDefinition) attribute).getValueTypes();
+                writer.writeStartElement(attribute.getXmlName());
+                for (AttributeDefinition valueType : valueTypes) {
+                    if(resourceModel.get(attribute.getName()).hasDefined(valueType.getName())) {
```

Comment:
Use the new ModelNode op:

```
if (resourceMode.hasDefined(attribute.getName(), valueType.getName())) {
```

That call has no chance of mutating a ModelNode where doing so isn't wanted, so I want us to start using it.